src/lib.rs

```diff
@@ -4,224 +4,204 @@ This is a cool-id-generator.
-use crate::animal::ANIMALS;
+#![no_std]
+
+#[macro_use]
+extern crate alloc;
+use alloc::string::String;
+
 use crate::adjective::ADJECTIVES;
+use crate::animal::ANIMALS;
+use crate::job::JOBS;
 use crate::names::NAMES;
-use crate::prefix::JOBS_PREFIX;
 use crate::prefix::ANIMAL_PREFIX;
+use crate::prefix::JOBS_PREFIX;
 
-use rand::thread_rng;
 use rand::seq::SliceRandom;
+use rand::thread_rng;
 
 mod adjective;
 mod animal;
-mod names;
 mod job;
+mod names;
 mod prefix;
 
-/// Creates ids in the format of {adjective}-{prefix}-{animal|job}-{name}
-/// e.g. "unpleasant-steampunk-poet-gerald"
-///
-/// Generates 1 billion combinations
-#[inline]
-pub fn get_id() -> String {
-    let mut rng = thread_rng();
-
-    let name = NAMES.choose(&mut rng).unwrap();
-    let adj1 = ADJECTIVES.choose(&mut rng).unwrap();
-    let animal = ANIMALS.choose(&mut rng).unwrap();
-    let job = JOBS.choose(&mut rng).unwrap();
-
-    // let combinations_animals = NAMES.len() * ADJECTIVES.len() * ANIMALS.len();
-    // let combinations_jobs = NAMES.len() * ADJECTIVES.len() * JOBS.len();
-
-    let is_animal_or_job = rand::random();
-    if is_animal_or_job {
-        // It's an animal!
-        let prefix = get_animal_prefix(&mut rng);
-        // println!("Total Combinations {:?}", combinations_animals * ANIMAL_PREFIX.len());
-        format!("{}{}-{}-{}", adj1, prefix, animal, name)
-    }else{
-        // It's a job!
-        let prefix = get_job_prefix(&mut rng);
-        // println!("Total Combinations {:?}", combinations_jobs * JOBS_PREFIX.len());
-        format!("{}{}-{}-{}", adj1, prefix, job, name)
-    }
-
-}
-
-/// Creates ids in the format of {name}-the-{adjective}-and-{adjective}-prefix-{animal|job}
-/// e.g. "unpleasant-steampunk-poet-gerald"
-///
-/// Generates 115 billion combinations
 #[inline]
-pub fn get_long_id() -> String {
+pub fn get_id(size: Size) -> String {
     let mut rng = thread_rng();
 
-    let name = NAMES.choose(&mut rng).unwrap();
-    let adj1 = ADJECTIVES.choose(&mut rng).unwrap();
-    let adj2 = ADJECTIVES.choose(&mut rng).unwrap();
-    let animal = ANIMALS.choose(&mut rng).unwrap();
-    let job = JOBS.choose(&mut rng).unwrap();
-
-    // let combinations_animals = NAMES.len() * ADJECTIVES.len() * ADJECTIVES.len() * ANIMALS.len();
-    // let combinations_jobs = NAMES.len() * ADJECTIVES.len() * ADJECTIVES.len() * JOBS.len();
-
-    let is_animal_or_job = rand::random();
-    if is_animal_or_job {
-        // It's an animal!
-        let prefix = get_animal_prefix(&mut rng);
-
-        // println!("Total Combinations {:?}", combinations_animals * ANIMAL_PREFIX.len());
-        format!("{}-the-{}-and-{}{}-{}", name, adj1, adj2, prefix, animal)
-    }else{
-        // It's a job!
-        let prefix = get_job_prefix(&mut rng);
-        // println!("Total Combinations {:?}", combinations_jobs * JOBS_PREFIX.len());
-        format!("{}-the-{}-and-{}{}-{}", name, adj1, adj2, prefix, job)
+    let animal_or_job = {
```

Comment:
the intermediate string can be avoided I think, by using `std::fmt::Arguments` or something similiar

Reply:
how does this look
```rust 
let animal_or_job:  (&&str, &&str) = {...}
...
format!("{}{}-{}-{}", adj1, animal_or_job.0, animal_or_job.1, name)
```

Reply:
looks good, probably easier to understand than arguments